src/relay/backend/build_module.cc

```diff
@@ -0,0 +1,728 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file relay/backend/build_module.cc
+ * \brief Code generation for TVM's graph runtime.
+ */
+
+#include <tvm/build_module.h>
+#include <tvm/relay/op.h>
+#include <tvm/relay/expr.h>
+#include <tvm/relay/attrs/nn.h>
+#include <tvm/relay/attrs/transform.h>
+#include <vector>
+#include <string>
+#include <memory>
+
+#include "utils.h"
+
+namespace tvm {
+namespace relay {
+namespace backend {
+
+/*!
+ * \brief Context name / index 
+ *        See: python/tvm/_ffi/runtime_ctypes.py
+ */
+struct ContextMap {
+  static const std::unordered_map<int, std::string> mask2str;
+  static const std::unordered_map<std::string, int> str2mask;
+  static std::string Mask2Str(int mask) {
+    CHECK_GT(mask2str.count(mask), 0) << "Unknown mask.";
+    return mask2str.at(mask);
+  }
+  static int Str2Mask(const std::string& str) {
+    CHECK_GT(str2mask.count(str), 0) << "Unknown context.";
+    return str2mask.at(str);
+  }
+  static std::unordered_map<int, std::string> _declare_mask2str() {
+    std::unordered_map<int, std::string> ret;
+    ret[1] = "cpu";
+    ret[2] = "gpu";
+    ret[4] = "opencl";
+    ret[5] = "aocl";
+    ret[6] = "sdaccel";
+    ret[7] = "valkan";
+    ret[8] = "metal";
+    ret[9] = "vpi";
+    ret[10] = "rocm";
+    ret[11] = "opengl";
+    ret[12] = "ext_dev";
+    return ret;
+  }
+  static std::unordered_map<std::string, int> _declare_str2mask() {
+    std::unordered_map<std::string, int> ret;
+    ret["llvm"] = 1;
+    ret["stackvm"] = 1;
+    ret["cpu"] = 1;
+    ret["c"] = 1;
+    ret["gpu"] = 2;
+    ret["cuda"] = 2;
+    ret["nvptx"] = 2;
+    ret["cl"] = 4;
+    ret["opencl"] = 4;
+    ret["aocl"] = 5;
+    ret["aocl_sw_emu"] = 5;
+    ret["sdaccel"] = 6;
+    ret["vulkan"] = 7;
+    ret["metal"] = 8;
+    ret["vpi"] = 9;
+    ret["rocm"] = 10;
+    ret["opengl"] = 11;
+    ret["ext_dev"] = 12;
+    return ret;
+  }
+};
+
+const std::unordered_map<int, std::string> ContextMap::mask2str =
+  ContextMap::_declare_mask2str();
+const std::unordered_map<std::string, int> ContextMap::str2mask =
+  ContextMap::_declare_str2mask();
+
+/*!
+ * \brief A data structure to map the names of specific optimizations to
+ *        numeric optimization levels
+ * 
+ */
+struct OptPassLevel {
+  static const std::unordered_map<std::string, int> _data;
```

Comment:
same here. You can just use braced list to initialize unordered_map.

Reply:
Because it is static const object...